app/code/Magento/Inventory/Model/StockSourceLink/Validator/AssignDefaultSourceToStocksValidator.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: Stepan Furman
+ * Date: 27.01.18
+ * Time: 16:12
+ */
+declare(strict_types=1);
+
+namespace Magento\Inventory\Model\StockSourceLink\Validator;
+
+use Magento\Framework\Validation\ValidationResult;
+use Magento\Framework\Validation\ValidationResultFactory;
+use Magento\Inventory\Model\StockSourceLink;
+
+class AssignDefaultSourceToStocksValidator implements StockSourceLinkValidatorInterface
+{
+    /**
+     * @var ValidationResultFactory
+     */
+    private $validationResultFactory;
+
+    /**
+     * @param ValidationResultFactory $validationResultFactory
+     */
+    public function __construct(ValidationResultFactory $validationResultFactory)
+    {
+        $this->validationResultFactory = $validationResultFactory;
+    }
+
+    /**
+     * @param StockSourceLink[] $links
+     * @return ValidationResult
+     */
+    public function validate(array $links): ValidationResult
+    {
+        $errors = [];
+        foreach ($links as $link) {
+            if ($link->getSourceCode() === 'default' && $link->getStockId() !== 1) {
+                $errors[] = __('Default Source can\'t be assigned to stock!');
```

Comment:
message should be more obvious, 
like "Just Default Source can be assigned to Default stock."